go/fn/object.go

```diff
@@ -759,6 +784,107 @@ func (o KubeObjects) Less(i, j int) bool {
 	return idStrI < idStrJ
 }
 
+func (o KubeObjects) String() string {
+	var elems []string
+	for _, obj := range o {
+		elems = append(elems, strings.TrimSpace(obj.String()))
+	}
+	return strings.Join(elems, "\n---\n")
+}
+
+// Select will return the subset of objects in KubeObjects such that f(object) returns 'true'.
+func (o KubeObjects) Select(f func(o *KubeObject) bool) KubeObjects {
+	var result KubeObjects
+	for _, obj := range o {
+		if f(obj) {
+			result = append(result, obj)
+		}
+	}
+	return result
+}
+
+// SelectByGvk will return the subset of objects that matches the provided GVK.
+func (o KubeObjects) SelectByGvk(apiVersion, kind string) KubeObjects {
```

Comment:
Nit: This isn't really selecting by GVK :-)  I would expect SelectByGVK to take a schema.GroupVersionKind.

Reply:
While I agree with the nit, we already have a `func (o *KubeObject) IsGVK(apiVersion, kind string) bool` [here](https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/68c3f914c65f2aff0f83b55efff171da54cc6743/go/fn/object.go#L587), so I'm trying to keep things consistent. If we're going to change it so that GVK strictly refers to group, version, and kind (rather than apiVersion and kind), we should do that everywhere. WDYT @mengqiy ?

Reply:
changed to use schema.GroupVersionKind instead of apiVersion

Reply:
I think the "schema.GroupVersionKind" refer to [k8s.io/apimachinery/pkg/runtime/schema ](https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime/schema#GroupVersionKind)